### PR TITLE
GOV/DOC: add section to docs on triaging and triage team

### PR DESCRIPTION
**Pull request by tacaswell**
## PR Summary

This adds documentation to add a "triage team" to our organization as discussed on today's call.  The purpose of this is to:

 - provide a low barrier to formally recognize users as contributing to the maintenance of Matplotlib  (by giving them power and responsibility)
 - empower more people to help where they are willing and able
 - to incentive more people to triage issues
 - (secondary) provide a ramp / pipeline to contributing PRs (and reviewing PRs!) for new contributors

This text is based on https://scikit-learn.org/dev/developers/bug_triaging.html  and modified to better fit Matplotlib.

Task that need to be done immediately  outside of a PR if we merge this:

 - create triage_team in the github ACL system (I think only me @dopplershift and @efiring can currently do that?) with triage rights on matplotlib/matplotlib
 
If this catches on we may want to:

 - add more formal method of requesting to join (ex a form), but for now while we figure out if this is even a good idea, lets keep it informal.
 -  add "triage team captain" or similar to the named roles in the governance docs, but again lets see if this catches on / is a good idea before we do that.

On the call I suggested 1 triage would be the threshold to be invited, but ended up with 2-3 just so that we have a bit more context to evaluate.  Being optomistic, I expect most of these requests / nominations to be fulfilled.  

sklearn has a slightly higher bar for inviting people to the triage team (https://scikit-learn.org/dev/governance.html#triage-team) and more ceremony, but I would rather have this be broader and less formal.  

We can deal with bad-faith actors if they come up and I expect a vast majority of the issues we will have will be miss-labeled or miss-milestoned issues which are almost trivially fixable.

### CODE REVIEW

File: doc/devel/triage.rst [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+
+
+Triage Team
+-----------
+
+
+If you would like to join the triage team:
+
+1. Correctly triage 2-3 issues.
+2. Ask someone with commit rights to recommend you to the triage team
```
Comment (story645) [inline doc/devel/triage.rst:71]: please link here to the list of people w/ commit rights (which separately, a lot of folk with commit rights aren't active)  & possibly the default person they should reach out to (which is why my bias is towards a form) 'cause I can imagine folk not wanting to bother someone w/ commit rights - if I was in this boat when I started contributing, I would have asked Ryan 'cause I already knew him. Possible add another sentence here is "for example, one of the contributors that you have interacted with on an issue" to make it clear the contributors are on board w/ it & who they should be asking.

File: doc/devel/triage.rst [hunk 2]
```diff
@@ -0,0 +1,214 @@
+
+.. _bug_triaging:
+
+Bug triaging and issue curation
+================================
+
+The `issue tracker <https://github.com/matplotlib/matplotlib/issues>`_
+is important to the communication in the project: it is where users to
+make feature requests and report bugs, developers identify major
```
Comment (story645) [inline doc/devel/triage.rst:9]: I am totally anti the dev/user split, especially since like we don't even have these breakdowns in the issue trackers. Folks with commit rights open tons of bugs and feature requests, discussions about priorities tend to happen amongst whoever is commenting in that issue. suggested instead:
> The `issue tracker <https://github.com/matplotlib/matplotlib/issues>`_ is important to communication in the project because it serves as the centralized location for making feature requests, reporting bugs, identifying major projects to work on, and discussing priorities.

File: doc/devel/triage.rst [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+closing issues that are not necessary.
+
+
+Triaging issues does not require any particular expertise in the
+internals of Matplotlib, is extremely valuable to the project, and we
+welcome anyone to participate in issue triage!  However, people who
+are not part of the Matplotlib organization do not have `permissions
+to change milestones, add labels, or close issue
+<https://docs.github.com/en/github/setting-up-and-managing-organizations-and-teams/repository-permission-levels-for-an-organization>`_.
+If you do not have enough permissions do to a step, please leave a
```
Comment (story645) [inline doc/devel/triage.rst:21]: kinda unclear what step means here - do you mean the tasks listed above? (milestone, label, close?)

File: doc/devel/triage.rst [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+Guidelines on submitting good issues can be found :ref:`here
+<submitting-a-bug-report>`.
+A third party can give useful feedback or even add
+comments on the issue.
+The following actions are typically useful:
+
+  - documenting issues that are missing elements to reproduce the problem
+    such as code samples
+
+  - suggesting better use of code formatting
```
Comment (story645) [inline doc/devel/triage.rst:37]: link to code format style here?

File: doc/devel/triage.rst [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+
+   The issue tracker is many people’s first interaction with the
+   matplotlib project itself, beyond just using the library. As such,
+   we want it to be a welcoming, pleasant experience.
+
+#. Is this a usage question? If so close it with a polite message.
+
+#. Is the necessary information provided?
+
+   If crucial information (the version of Python, the version of
```
Comment (story645) [inline doc/devel/triage.rst:137]: this is the template - might be worth linking to the templates here - did they fill out the template- especially as these are variable depending on issue type.

**Review by story645 — CHANGES_REQUESTED** [review]

File: doc/devel/triage.rst [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+  - verifying that the issue is reproducible
+
+  - classify the issue as a feature request, a long standing bug or a
+    regression
+
+.. topic:: Fruitful discussions
+
+   Online discussions may be harder than it seems at first glance, in
+   particular given that a person new to open-source may have a very
+   different understanding of the process than a seasonned maintainer.
```
Comment (QuLogic) [inline doc/devel/triage.rst:56]:
```suggestion
   different understanding of the process than a seasoned maintainer.
```

File: doc/devel/triage.rst [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+   (publicly or privately).
+3. Responsibly exercise you new power!
+
+Anyone with commit or triage rights may also nominate a user to be
+invited to join the triage team.
+
+
+
+Triaging operations for members of the core and triage teams
+-------------------------------------------------------------
```
Comment (QuLogic) [inline doc/devel/triage.rst:81]:
```suggestion
Triaging operations for members of the core and triage teams
------------------------------------------------------------
```

File: doc/devel/triage.rst [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+- Update labels for issues and PRs: see the list of
+  the `available github labels
+  <https://github.com/matplotlib/matplotlib/labels>`_.
+
+- Triage issues:
+
+  - **reproduce the issue**, if the posted code is a bug label the issue
+    with "status: confirmed bug".
+
+  - **identify regressions**, determine if the reported bug used to
```
Comment (QuLogic) [inline doc/devel/triage.rst:95]: And/or attempt a bisect?

File: doc/devel/triage.rst [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+
+A typical workflow for triaging issues
+--------------------------------------
+
+The following workflow [1]_ is a good way to approach issue triaging:
+
+#. Thank the reporter for opening an issue
+
+   The issue tracker is many people’s first interaction with the
+   matplotlib project itself, beyond just using the library. As such,
```
Comment (QuLogic) [inline doc/devel/triage.rst:130]:
```suggestion
   Matplotlib project itself, beyond just using the library. As such,
```

File: doc/devel/triage.rst [hunk 10]
```diff
@@ -0,0 +1,214 @@
...
+   the original poster to provide the information.
+
+#. Is the issue minimal and reproducible?
+
+   For bug reports, we ask that the reporter provide a minimal
+   reproducible example. See `this useful post
+   <https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports>`_
+   by Matthew Rocklin for a good explanation. If the example is not
+   reproducible, or if it's clearly not minimal, feel free to ask the reporter
+   if they can provide and example or simplify the provided one.
```
Comment (QuLogic) [inline doc/devel/triage.rst:148]:
```suggestion
   if they can provide an example or simplify the provided one.
```

File: doc/devel/triage.rst [hunk 11]
```diff
@@ -0,0 +1,214 @@
...
+   Do acknowledge that writing minimal reproducible examples is hard work.
+   If the reporter is struggling, you can try to write one yourself.
+
+   If a reproducible example is provided, but you see a simplification,
+   add your simpler reproducible example.
+
+   If you can not reproduce the issue, please report that along with your
+   OS, Pyhthon, and Matplotlib versions.
```
Comment (QuLogic) [inline doc/devel/triage.rst:156]:
```suggestion
   OS, Python, and Matplotlib versions.
```

File: doc/devel/triage.rst [hunk 12]
```diff
@@ -0,0 +1,214 @@
...
+
+   If we need more information from either this or the previous step
+   please label the issue with "status: needs clarification".
+
+#. Is this a regression?
+
+   While we strive for a bug-free library, regressions are the highest
+   priority.  If we have broken user-code that *used to* work, we should
+   fix that in the next patch release!
```
Comment (QuLogic) [inline doc/devel/triage.rst:165]: Attempt to bisect here (if reproducer is simple, known good version exists, etc.)?

File: doc/devel/triage.rst [hunk 13]
```diff
@@ -0,0 +1,214 @@
...
+   or an edit to the original post.
+
+   Label the closed issue with "status: duplicate"
+
+#. Make sure that the title accurately reflects the issue. If you have the
+   necessary permissions edit it yourself if it's not clear.
+
+#. Add the relevant labels, such as "Documentation" when the issue is
+   about documentation, "Bug" if it is clearly a bug, "Enhancement" if it
+   is an enhancement request, ...
```
Comment (QuLogic) [inline doc/devel/triage.rst:186]: We don't have an Enhancement label any more.
```suggestion
   about documentation, "Bug" if it is clearly a bug, "New feature" if it
   is a new feature request, ...
```

File: doc/devel/triage.rst [hunk 14]
```diff
@@ -0,0 +1,214 @@
...
+
+   If the issue is clearly defined and the fix seems relatively
+   straightforward, label the issue as “Good first issue” (and
+   possibly a description of the fix or a hint as to where in the
+   code base to look to get started).
+
+   An additional useful step can be to tag the corresponding module e.g.
+   the "GUI/Qt" when relevant.
```
Comment (QuLogic) [inline doc/devel/triage.rst:194]:
```suggestion
   the "GUI/Qt" label when relevant.
```

File: doc/devel/triage.rst [hunk 15]
```diff
@@ -0,0 +1,214 @@
...
+
+
+.. [1] Adapted from the pandas project `maintainers guide
+       <https://dev.pandas.io/docs/development/maintaining.html>`_ and
+       `the sckit-learn project
```
Comment (QuLogic) [inline doc/devel/triage.rst:199]:
```suggestion
       `the scikit-learn project
```

File: doc/devel/triage.rst (hunk 6 above)
Comment (tacaswell) [reply doc/devel/triage.rst:56]: This was un-edited from the sklean text, going to open a PR with them too :-p

File: doc/devel/triage.rst (hunk 2 above)
Comment (tacaswell) [reply doc/devel/triage.rst:9]: I agree with the dev/user split being bad in general.  This particular wording didn't bother me too much as different people in the community do have different responsibilities, but have taken your wording here.

File: doc/devel/triage.rst (hunk 4 above)
Comment (tacaswell) [reply doc/devel/triage.rst:37]: I think this is aimed at improving mark-down usage not the Python, clarified.

File: doc/devel/triage.rst (hunk 12 above)
Comment (tacaswell) [reply doc/devel/triage.rst:176]: Added paragraph below (so GH did not notice).

File: doc/devel/triage.rst [hunk 16]
```diff
@@ -0,0 +1,229 @@
...
+
+   While we strive for a bug-free library, regressions are the highest
+   priority.  If we have broken user-code that *used to* work, we should
+   fix that in the next patch release!
+
+   Try to determine when the regression happened by running the
+   reproduction code against older versions of Matplotlib.  This can
+   be done by released versions of Matplotlib (to get the version it
+   last worked in) or by using `git bisect
+   <https://git-scm.com/docs/git-bisect>`_ . to find first commit
```
Comment (QuLogic) [inline doc/devel/triage.rst:178]:
```suggestion
   <https://git-scm.com/docs/git-bisect>`_ to find the first commit
```

File: doc/devel/triage.rst [hunk 17]
```diff
@@ -0,0 +1,229 @@
...
+
+- Triage issues:
+
+  - **reproduce the issue**, if the posted code is a bug label the issue
+    with "status: confirmed bug".
+
+  - **identify regressions**, determine if the reported bug used to
+    work as expected in a recent version of Matplotlib and if so
+    determine the last working version.  Regression should be
+    milestone for the next bug-fix release and may be labeled as
```
Comment (QuLogic) [inline doc/devel/triage.rst:103]:
```suggestion
    determine the last working version.  Regressions should be
    milestoned for the next bug-fix release and may be labeled as
```

File: doc/devel/triage.rst [hunk 18]
```diff
@@ -0,0 +1,229 @@
...
+
+Bug triaging and issue curation
+===============================
+
+The `issue tracker <https://github.com/matplotlib/matplotlib/issues>`_
+is important to communication in the project because it serves as the
+centralized location for making feature requests, reporting bugs,
+identifying major projects to work on, and discussing priorities.  For
+this reason, it is important to curate the issue list, adding labels
+to issues and closing issues that are not necessary.
```
Comment (story645) [inline doc/devel/triage.rst:12]:
```suggestion
to issues and closing issues that are resolved or unresolvable
```re

File: doc/devel/triage.rst [hunk 19]
```diff
@@ -0,0 +1,229 @@
...
+
+
+Triage Team
+-----------
+
+
+If you would like to join the triage team:
+
+1. Correctly triage 2-3 issues.
+2. Ask someone with on the `triage team
```
Comment (story645) [inline doc/devel/triage.rst:73]:
```suggestion
2. Ask someone on the `triage team
```

File: doc/devel/triage.rst [hunk 20]
```diff
@@ -0,0 +1,229 @@
...
+
+
+Triaging operations for members of the core and triage teams
+------------------------------------------------------------
+
+In addition to the above, members of the core team and the triage team
+can do the following important tasks:
+
+- Update labels for issues and PRs: see the list of
+  the `available github labels
```
Comment (story645) [inline doc/devel/triage.rst:92]:
```suggestion
 `available github labels
```

File: doc/devel/triage.rst [hunk 21]
```diff
@@ -0,0 +1,229 @@
...
+    least a week) to add extra information
+
+
+
+.. topic:: Closing issues: a tough call
+
+    When uncertain on whether an issue should be closed or not, it is
+    best to strive for consensus with the original poster, and possibly
+    to seek relevant expertise. However, when the issue is a usage
+    question, or when it has been considered as unclear for many years it
```
Comment (story645) [inline doc/devel/triage.rst:124]:
```suggestion
    question or has been considered as unclear for many years, then it
```

**Review by story645 — APPROVED** [review]

**Comment by jklymak** [comment]
@dopplershift @timhoffm @efiring   This is on the agenda for today, but if you guys are OK with adding a "triage" layer to the Membership then you can hit approve.  Thanks!

File: doc/devel/triage.rst [hunk 22]
```diff
@@ -0,0 +1,230 @@
...
+
+If you would like to join the triage team:
+
+1. Correctly triage 2-3 issues.
+2. Ask someone on the `triage team
+   <https://github.com/orgs/matplotlib/teams/triageteam>`_ (publicly
+   or privately) to recommend you to the triage team .  If you worked
+   with someone on the issue triaged, they would be a good person to
+   ask.
+3. Responsibly exercise you new power!
```
Comment (efiring) [inline doc/devel/triage.rst:77]: you -> your

**Review by dopplershift — APPROVED** [review]

**Review by efiring — APPROVED** [review]
Just one typo.
Looks good to me.

**Comment by tacaswell** [comment]
I will merge this when docs pass CI.

**Comment by tacaswell** [comment]
Testing @matplotlib/triageteam notification.
